### PR TITLE
feat: thorchain savers metadata resolver

### PR DESCRIPTION
**Pull request by gomesalexandre**
## Description

This PR resolves and populates `DefiProvider.ThorchainSavers` opportunities.

Notes:

- APR cannot yet be fetched - to be fixed in a follow-up web PR after we upgrade our Midgard proxy:

https://github.com/shapeshift/web/blob/a369170145b78979ee9f9482664a50b75e5fe858/src/state/slices/opportunitiesSlice/resolvers/thorchainsavers/index.ts#L103-L105

- Ethereum opportunity doesn't yet appear, figure out why and fix it in a follow-up web PR

## Notice

- [x] Have you followed the guidelines in our [Contributing]('https://github.com/shapeshift/web/blob/develop/CONTRIBUTING.md) guide?
- [x] Have you checked to ensure there aren't other open [Pull Requests](https://github.com/shapeshift/web/pulls) for the same update/change?

## Pull Request Type

- [ ] :bug: Bug fix (Non-breaking Change: Fixes an issue)
- [ ] :hammer_and_wrench: Chore (Non-breaking Change: Doc updates, pkg upgrades, typos, etc..)
- [x] :nail_care: New Feature (Breaking/Non-breaking Change)

## Issue (if applicable)

- Tackles https://github.com/shapeshift/web/issues/3584

## Risk

In theory, none given the resolver short circuits with the flag off right at entrypoint - in practice, see ops testing notes for actual risks given the slight refactor

### Engineering

- Toggle `REACT_APP_FEATURE_SAVERS_VAULTS`
- Ensure Thorchain Savers metadata is populated in store

### Operations

- Logic has been slightly revamped because tsc is drunk - clear your persisted opportunities cache and confirm no regressions

## Screenshots (if applicable)

image

### CODE REVIEW

**Comment by gomesalexandre** [comment]
Current dependencies on/for this PR:
* develop
  * **PR #3583** <a href="https://app.graphite.dev/github/pr/shapeshift/web/3583" target="_blank">Graphite</a>  👈

This comment was auto-generated by [Graphite](https://app.graphite.dev/github/pr/shapeshift/web/3583?utm_source=stack-comment).

File: src/features/defi/providers/thorchain-savers/components/ThorchainSaversManager/Overview/ThorchainSaversOverview.tsx [hunk 1]
```diff
@@ -79,7 +79,8 @@ export const ThorchainSaversOverview: React.FC<OverviewProps> = ({
   const yearnInvestor = useMemo(() => getYearnInvestor(), [])
   const translate = useTranslate()
   const { query } = useBrowserRouter<DefiQueryParams, DefiParams>()
-  const { chainId, contractAddress, assetReference } = query
+  const { chainId, contractAddress, assetReference, assetNamespace } = query
+  console.log({ chainId, contractAddress, assetReference, assetNamespace })
```
Comment (0xdef1cafe) [inline src/features/defi/providers/thorchain-savers/components/ThorchainSaversManager/Overview/ThorchainSaversOverview.tsx:83]: delet

File: src/state/slices/opportunitiesSlice/resolvers/thorchainsavers/index.ts [hunk 2]
```diff
@@ -33,3 +55,72 @@ export const thorchainSaversOpportunityIdsResolver = async (): Promise<{
...
+    const opportunityId = assetId as OpportunityId
+
+    // Thorchain is slightly different from other opportunities in that there is no contract address for the opportunity
+    // The way we represent it, the opportunityId is both the opportunityId/assetId and the underlyingAssetId
+    // That's an oversimplification, as this ties a native AssetId e.g btcAssetId or ethAssetId, to a Savers opportunity
+    // If we were to ever support another native asset staking opportunity e.g Ethereum 2.0 consensus layer staking
+    // we would need to revisit this by using generic keys as an opportunityId
+    const asset = selectAssetById(state, assetId)
+    const underlyingAsset = selectAssetById(state, assetId)
+    const marketData = selectMarketDataById(state, assetId)
```
Comment (0xdef1cafe) [inline src/state/slices/opportunitiesSlice/resolvers/thorchainsavers/index.ts:94]: const underlyingAsset = asset

**Review by 0xdef1cafe — APPROVED** [review]
NOICE

**Pull request merged**